### PR TITLE
Validate components against layout schema

### LINKED ISSUE #1512: Validate individual components against JsonSchema and produce warnings in DevTools
Extracted from #648. We should bring a copy of the component JsonSchema into app-frontend, and validate all components in a layout file against that, spitting out errors in DevTools if any.

### PR DESCRIPTION
**Pull request by bjosttveit**
## Description

Validate component configuration (from redux `formLayout/layouts`) against `layout.schema.v1.json`.
- Copies schema folder to dist on build so it will be included in CDN alongside `altinn-app-frontend.js`.
- Serve schemas in webpack-dev-server for local development.
- Added a `useLayoutValidation` hook to dev tools
  - Fetches the schema from CDN
  - It removes references to `expression.schema.v1.json` and replaces it with `anyOf[<underlying type>, array]`. This is to avoid validating actual expressions using schema since it does a rather poor job and generates a ton of errors with little value.
  - It is enabled when in `dev` (localtest, tt02, preview) but not in prod unless dev-tools is opened up.
  - Logs the errors per component to dev tools.
- Checks the type of group component first (repeating, non-repeating, likert...) and only validates against the specific schema for that type.
- Validates Grid cells individually and uses a similar method as groups to determine more specific schema pointers to use.
- Added a unit test (thx @olemartinorg) to validate against all apps in tt02 using the `ALTINN_ALL_APPS_DIR` variable, this was mainly to ensure that it only produces useful messages
- Errors show up in Layout overview in dev tools (thx @olemartinorg). Also updated the styling a bit to format errors more nicely.

Here are some examples of what the resulting errors from `frontend-test` looks like:

image

image

## Related Issue(s)

- closes #1512 

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [x] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [x] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: webpack.config.development.js [hunk 1]
```diff
@@ -69,12 +71,15 @@ module.exports = {
...
+    headers: { 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Headers': '*' },
     client: {
       overlay: {
         errors: true,
         warnings: false,
       },
     },
+    static: {
+      directory: path.join(__dirname, 'dist'),
+    },
```
[the remark this answers is not in the document]
Comment (bjosttveit) [reply webpack.config.development.js:83]: As long as I can get the same url path sure

File: src/features/devtools/utils/layoutValidation.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+import { duplicateStringFilter } from 'src/utils/stringHelper';
+import type { CompOrGroupExternal, ILayouts } from 'src/layout/layout';
+
+export const LAYOUT_SCHEMA_NAME = 'layout.schema.v1.json';
+
+const COMPONENT_POINTER = '#/definitions/AnyComponent';
+const NON_REPEATING_GROUP_POINTER = '#/definitions/CompGroupNonRepeating';
+const NON_REPEATING_GROUP_PANEL_POINTER = '#/definitions/CompGroupNonRepeatingPanel';
+const REPEATING_GROUP_POINTER = '#/definitions/CompGroupRepeating';
+const REPEATING_GROUP_LIKERT_POINTER = '#/definitions/CompGroupRepeatingLikert';
```
[the remark this answers is not in the document]
Comment (bjosttveit) [reply src/features/devtools/utils/layoutValidation.ts:21]: Only thing is I dont necessarily have any nodes to work with here, as I by definition have to work on the external config, and as far as I know the nodes don't have access to their external config currently?

File: src/features/devtools/utils/layoutValidation.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
+/* eslint-disable no-case-declarations */
```
[the remark this answers is not in the document]
Comment (bjosttveit) [reply src/features/devtools/utils/layoutValidation.ts:1]: I had some difficulties running the validation as a node script without separating most of the logic into its own file

File: src/features/devtools/utils/layoutValidation.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (olemartinorg) [reply src/features/devtools/utils/layoutValidation.ts:21]: Yeah, they don't - but I wasn't thinking of the Node classes (they're for the instances of each component), but the Def classes, the one in `src/layout/Group/index.tsx`

**Review by olemartinorg — APPROVED** [review]
🥳

**Review by olemartinorg — APPROVED** [review]

**Pull request merged**